src/composables/useMarkdownActions.ts

```diff
@@ -482,6 +495,88 @@ export default function useMarkdownActions(
     }
   }
 
+  /**
+   * Insert a markdown image at the current cursor position.
+   * @returns {Promise<void>}
+   */
+  const insertImage = async (): Promise<void> => {
+    try {
+      const textarea = getTextarea()
+
+      if (!textarea) {
+        return
+      }
+
+      // Update the selected text object
+      getTextSelection()
+
+      // Get the text before and after the cursor
+      const startText = rawMarkdown.value.substring(0, selectedText.start)
+      const endText = rawMarkdown.value.substring(selectedText.end)
+      let newContent: string = ''
+
+      // If text is selected, check the type of selected text and insert the link template around it
+      if (selectedText.text.length !== 0) {
+        // If the user tries to click the button twice (with `url` selected) exit early
+        if (selectedText.text === 'url' && startText.endsWith('(') && endText.startsWith(')')) {
+          await focusTextarea()
+          return
+        }
+
+        // Check if the selected text is a URL
+        const isUrl = /^http(s)?:\/\//.test(selectedText.text)
+        // Prepare the content
+        newContent = isUrl ? MARKDOWN_TEMPLATE_IMAGE.replace(/alt/, '').replace(/url/, selectedText.text) : MARKDOWN_TEMPLATE_IMAGE.replace(/alt/, selectedText.text)
+
+        // Update the markdown
+        rawMarkdown.value = startText + newContent + endText
+
+        // Always focus back on the textarea
+        await focusTextarea()
+
+        // Set the cursor position
+        if (isUrl) {
+          textarea.selectionEnd = selectedText.start + 2
+        } else {
+          textarea.selectionStart = startText.length + selectedText.text.length + 3
+          textarea.selectionEnd = startText.length + selectedText.text.length + 6
+        }
+      } else {
+        // No text is selected
+
+        // If the user tries to click the button twice (with the cursor in between the brackets) exit early
+        if (startText.endsWith(MARKDOWN_TEMPLATE_IMAGE.split('alt')[0]) && /^\]\((.*)+\)/.test(endText)) {
+          await focusTextarea()
+          return
+        }
+
+        // Prepare the content
+        newContent = MARKDOWN_TEMPLATE_IMAGE.replace(/alt/, '')
```

Comment:
suggestion: let's not remove the `alt` text so it's obvious what it is. Can you make the same change on line `472` so that the `link` is now this: `newContent = MARKDOWN_TEMPLATE_LINK`
```suggestion
        newContent = MARKDOWN_TEMPLATE_IMAGE
```

Reply:
changed